src/scripts/EasternKingdoms/Karazhan/boss_chess_event.cpp

```diff
@@ -0,0 +1,1838 @@
+
+            if (m_instance->GetPlayerTeam() == HORDE)
+            {
+                switch (urand(0, 2))
+                {
+                case 0: DoPlaySoundToSet(medivh, SOUND_ID_LOSE_PAWN_PLAYER_1); break;
+                case 1: DoPlaySoundToSet(medivh, SOUND_ID_LOSE_PAWN_PLAYER_2); break;
+                case 2: DoPlaySoundToSet(medivh, SOUND_ID_LOSE_PAWN_PLAYER_3); break;
+                }
+            }
+            else
+            {
+                switch (urand(0, 2))
+                {
+                case 0: DoPlaySoundToSet(medivh, SOUND_ID_LOSE_PAWN_MEDIVH_1); break;
+                case 1: DoPlaySoundToSet(medivh, SOUND_ID_LOSE_PAWN_MEDIVH_2); break;
+                case 2: DoPlaySoundToSet(medivh, SOUND_ID_LOSE_PAWN_MEDIVH_3); break;
+                }
+            }
+
+            m_instance->DoMoveChessPieceToSides(SPELL_TRANSFORM_GRUNT, FACTION_ID_CHESS_HORDE);
+        }
+
+        uint32 DoCastPrimarySpell() override
+        {
+            if (Unit* target = GetTargetByType(TARGET_TYPE_RANDOM, 8.0f, M_PI / 12))
+            {
+                DoCast(me, SPELL_VICIOUS_STRIKE);
+
+                // reset timer based on spell values
+                const SpellEntry* spell = sSpellStore.LookupEntry(SPELL_VICIOUS_STRIKE);
+                return spell->RecoveryTime ? spell->RecoveryTime : spell->CategoryRecoveryTime;
+            }
+
+            return 5000;
+        }
+
+        uint32 DoCastSecondarySpell() override
+        {
+            if (Unit* target = GetTargetByType(TARGET_TYPE_RANDOM, 8.0f))
+            {
+                DoCast(me, SPELL_WEAPON_DEFLECTION);
+
+                // reset timer based on spell values
+                const SpellEntry* spell = sSpellStore.LookupEntry(SPELL_WEAPON_DEFLECTION);
+                return spell->RecoveryTime ? spell->RecoveryTime : spell->CategoryRecoveryTime;
+            }
+
+            return 5000;
+        }
+    };
+
+    bool OnGossipHello(Player* player, Creature* creature)
+    {
+        if (player->HasAura(SPELL_RECENTLY_IN_GAME) || creature->HasAura(SPELL_CONTROL_PIECE))
+            return true;
+
+        if (InstanceScript* instance = (InstanceScript*)creature->GetInstanceScript())
+        {
+            if (instance->GetData(DATA_CHESS) == IN_PROGRESS && player->GetTeamId() == HORDE)
+                player->PlayerTalkClass->GetGossipMenu().AddMenuItem(GOSSIP_ICON_CHAT, GOSSIP_ITEM_ORC_GRUNT, GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+        }
+
+        player->SEND_GOSSIP_MENU(GOSSIP_MENU_ID_GRUNT, creature->GetGUID());
+        return true;
+    }
+    bool OnGossipSelect(Player* player, Creature* creature, uint32 sender, uint32 action) override
+    {
+        return OnGossipSelectGeneric(player, creature, sender, action);
+    }
+};
+
+/*######
+## npc_water_elemental
+######*/
+
+class npc_water_elemental : public CreatureScript
+{
+public:
+    npc_water_elemental() : CreatureScript("npc_water_elemental") { }
```

Comment:
There is already a creature in the database called npc_water_elemental try change the script name to something like npc_kara_water_elemental